eth_utils/abi.py

```diff
@@ -42,19 +95,592 @@ def collapse_if_tuple(abi: Dict[str, Any]) -> str:
+        abi_inputs = None
+    elif abi["type"] == "constructor":
+        name = abi["type"]
+        abi_inputs = abi["inputs"]
+    elif abi["type"] == "event":
+        name = abi["name"]
+        abi_inputs = abi["inputs"]
+    else:
+        name = str(abi.get("name", abi["type"]))
+        abi_inputs = abi.get("inputs")
+
+    return f"{name}({_abi_inputs_types(abi_inputs)})"
+
+
+def _filter_by_type(_type: str, contract_abi: ABI) -> List[ABIElement]:
+    return [abi for abi in contract_abi if abi["type"] == _type]
+
+
+def _filter_by_name(name: str, contract_abi: ABI) -> List[ABIElement]:
+    return [
+        abi
+        for abi in contract_abi
+        if (
+            abi["type"] not in ("fallback", "constructor", "receive")
+            and abi.get("name", "") == name
+        )
+    ]
+
+
+def _filter_by_encodability(
+    abi_codec: Any,
+    args: Sequence[Any],
+    kwargs: Dict[str, Any],
+    contract_abi: ABI,
+) -> List[ABIFunction]:
+    return [
+        abi
+        for abi in contract_abi
+        if abi["type"] == "function"
+        and _check_if_arguments_can_be_encoded(abi, abi_codec, args, kwargs)
+    ]
+
+
+def _filter_by_argument_name(
+    argument_names: Collection[str], contract_abi: ABI
+) -> List[ABIElement]:
+    return [
+        abi
+        for abi in contract_abi
+        if set(argument_names).intersection(get_abi_input_names(abi))
+        == set(argument_names)
+    ]
+
+
+def _get_abi_if_input_size(abi_element: ABIElement, num_arguments: int) -> bool:
+    if (
+        "inputs" not in abi_element
+        or abi_element["type"] == "fallback"
+        or abi_element["type"] == "receive"
+    ):
+        # only return True for fallback and receive if 0 arguments expected
+        return num_arguments == 0
+    elif abi_element["type"] == "function" or abi_element["type"] == "constructor":
+        return len(abi_element["inputs"]) == num_arguments
+
+    return False
+
+
+def _filter_by_argument_count(
+    num_arguments: int, contract_abi: ABI
+) -> List[ABIElement]:
+    return [abi for abi in contract_abi if _get_abi_if_input_size(abi, num_arguments)]
+
+
+def _check_if_arguments_can_be_encoded(
+    function_abi: ABIFunction,
+    abi_codec: Any,
+    args: Sequence[Any],
+    kwargs: Dict[str, Any],
+) -> bool:
+    try:
+        arguments = _merge_args_and_kwargs(function_abi, args, kwargs)
+    except TypeError:
+        return False
+
+    if len(function_abi.get("inputs", [])) != len(arguments):
+        return False
+
+    try:
+        types, aligned_args = _get_aligned_abi_inputs(function_abi, arguments)
+    except TypeError:
+        return False
+
+    return all(
+        abi_codec.is_encodable(_type, arg) for _type, arg in zip(types, aligned_args)
+    )
+
+
+def _merge_args_and_kwargs(
+    function_abi: ABIFunction, args: Sequence[Any], kwargs: Dict[str, Any]
+) -> Tuple[Any, ...]:
+    """
+    Takes a list of positional args (``args``) and a dict of keyword args
```

Comment:
I'm not seeing any of the docstrings show up in the RTD builds. I think you need a placeholder in the `utilites.rst` to have it expand, like:

```
.. automodule:: eth_utils.abi
    :members:
    :undoc-members:
    :show-inheritance:
```

But that expands the ones that are already there too. Might be cleaner to rewrite those as docstrings?